src/pip/_internal/format_control.py

```diff
@@ -0,0 +1,64 @@
+from pip._vendor.packaging.utils import canonicalize_name
+
+
+class FormatControl(object):
+    """This class has two fields, no_binary and only_binary.
+    If a field is falsy, it isn't set. If it is {':all:'}, it should match all
```

Comment:
Is there any situation when the fields are not a set and it has a significant meaning?

IOW, is `None` for these fields treated differently from an empty set? If not, let's restrict them to only being sets and assert that first thing in `__init__`.